### ISSUE TITLE
Generate XML Documentation

### ISSUE THREAD

**Issue by YanYas** [MyCaffe/MyCaffe#8]
Your code is well documented, but the info isn't currently available in the Nuget. With the next release a [generated XML](https://blogs.msdn.microsoft.com/msgulfcommunity/2014/04/22/creating-documentation-in-c-using-visual-studio-and-sandcastle/) file would be very much appreciated.

Thanks again

Yy

**Comment by ZoroDerVonCodier** [comment]
Hi @YanYas ,

We use doxygen to generate the documentation - to view the online documents, please see: https://www.signalpop.com/onlinehelp/mycaffe/html/index.html.

Also, you may find other helpful documents at:
https://www.signalpop.com/help

Hope this helps!

**Comment by YanYas** [comment]
Hi @ZoroDerVonCodier, 
So far I've used the doxygen doc for a high-level understanding and the source code for low level.
With the XML doc the information is presented with Intellisense. The IDE I use does this in its own way. At times I don't know what function I'm looking for so looking through the functions listed in this way means the flow isn't broken navigating a separate document. But it requires the xml file to do so.

![ListInfo](https://user-images.githubusercontent.com/9497052/71125025-10691580-21de-11ea-887e-f46ddddc2af2.gif)

**Comment by ZoroDerVonCodier** [comment]
Hi @YanYas ,

Good points all around - we will work on getting this into the next release.  Thank you for making the suggestion.  I am already seeing that adding this support is helping find several hard to find doxygen bugs that we have been chasing down!  Our next release is scheduled for end of year or early January.

**Comment by YanYas** [comment]
Thanks @ZoroDerVonCodier,

I'll close this at the next release, but thanks for your hard work.

**Comment by ZoroDerVonCodier** [comment]
Hi @YanYas ,

We have successfully added the XML documentation to MyCaffe and MyCaffe.basecode and were able to do so in a doxygen friendly way, so now we will have both.  By next release we will complete the other modules.  These XML files will be bundled in the next drop's nuget package and within the MyCaffe install.

During this process, we discovered a serious bug in our LaTex function generation, which we have fixed.  For that reason, we are also updating the online documentation today for the new version is much more accurate and has previous navigation bugs fixed. 

The latest online help can be found at: https://www.signalpop.com/onlinehelp/mycaffe/html/index.html

Again, thanks for suggesting the XML documentation for adding this helped us resolve a lot of other help related issues as well!
